datafusion/optimizer/src/simplify_expressions/simplify_exprs.rs

```diff
@@ -122,14 +123,21 @@ impl SimplifyExpressions {
 
         // Preserve expression names to avoid changing the schema of the plan.
         let name_preserver = NamePreserver::new(&plan);
-        plan.map_expressions(|e| {
-            let original_name = name_preserver.save(&e);
-            let new_e = simplifier
-                .simplify(e)
-                .map(|expr| original_name.restore(expr))?;
+        let mut rewrite_expr = |expr: Expr| {
+            let name = name_preserver.save(&expr);
+            let expr = simplifier.simplify(expr)?;
             // TODO it would be nice to have a way to know if the expression was simplified
             // or not. For now conservatively return Transformed::yes
-            Ok(Transformed::yes(new_e))
+            Ok(Transformed::yes(name.restore(expr)))
+        };
+
+        plan.map_expressions(|expr| {
+            // Preserve the aliasing of grouping sets.
```

Comment:
tis seems like it does the opposite (doesn't preserve the original names 🤔 )

Reply:
What makes you think so?

Reply:
Grouping sets needs to maintain the alias of the _children expressions_ as the field names needs to be based on that rather than the outer expression.

Reply:
> What makes you think so?

I was thinking that the  code skips calling `rewrite_expr` (which calls name preserver) for the GroupingSets (so thus does not preserve the aliases of the `Expr::Grouping` itself

To be clear I think the code in PR looks good to me, I am just discussing if we can make the comment better

Reply:
> What makes you think so?

I was thinking that the  code skips calling `rewrite_expr` (which calls name preserver) for the GroupingSets (so thus does not preserve the aliases of the `Expr::Grouping` itself

Reply:
Ok, I meant it in the sense that `Expr::GroupinSets` is a container of the actual grouping sets (which are a `Vec<Expr>` or `Vec<Vec<Expr>>`) and we want to preserve their names. But I'm happy to reword it however you like.

Reply:
no need to reword if I am the only one confused :)